src/overcooked_ai_py/mdp/overcooked_mdp.py

```diff
@@ -2025,32 +2083,43 @@ def make_closest_feature(idx, name, locations):
 
                 all_features["p{}_wall_{}".format(i, direction)] = [0] if feat == ' ' else [1]
 
+        # Convert feature dict created above into np array for each player
+        # NOTE: This is the portion that assumes n=2 players
         features_np = {k: np.array(v) for k, v in all_features.items()}
-
+        features_np = {k : np.expand_dims(v, 0) if not v.shape else v for k, v in features_np.items()}
```

Comment:
Could you please explain a bit what's actually going on here?

Reply:
This was there to convert 'scalar' features into single dimension numpy arrays. For example, the feature `p0_closest_soup_num_onions` might be set to the value of 3 in the feature dictionary. Calling `np.array(3)` returns a zero dimension array, so the `expand_dims` was necessary to get it into a shape compatible with all the other 1D array features.

However, I changed the code to avoid this confusion by setting `p0_closest_soup_num_onions = [3]` (in the above example) from the beginning

Reply:
I also caught a bug in this function, so now the code here is a little outdated